Add initial production release of Cloudsmith integration (v1.0.0) (#2675)
DataDog/integrations-extras: cloudsmith/README.md

Reviewer: Under ## Installation, replace "<INTEGRATION_VERSION>" with "1.0.0"

> ## Overview

Reviewer: This overview needs to be rewritten to answer the following:

1. What does your company/platform/product do?
2. How does this integration enhance the customers’ workflows or observability?
3. What data is collected from your product, and which Datadog data types is it mapped to (e.g., logs, metrics, events)?

### Metrics

> This integration provides the following metrics:
> 
> - `cloudsmith.storage.percentage_used`: Percentage of storage used.
> - `cloudsmith.storage.used_bytes`: Total storage used in bytes.
> - `cloudsmith.storage.peak_bytes`: Peak storage usage in bytes.
> - `cloudsmith.bandwidth.percentage_used`: Percentage of bandwidth used.
> - `cloudsmith.bandwidth.used_bytes`: Total bandwidth used in bytes.
> - `cloudsmith.members.total`: Total number of members in the organization.
> - `cloudsmith.members.owners`: Total number of owners.
> - `cloudsmith.members.managers`: Total number of managers.
> - `cloudsmith.members.readers`: Total number of readers.
> - `cloudsmith.tokens.total`: Number of entitlement tokens configured.
> - `cloudsmith.tokens.active`: Number of active entitlement tokens.
> - `cloudsmith.vulnerabilities.total`: Number of vulnerabilities found.
> - `cloudsmith.license_policy_violations.total`: Number of license policy violations detected.
> - `cloudsmith.vulnerability_policy_violations.total`: Number of vulnerability policy violations detected.
> 
> See [metadata.csv][8] for the full list and detailed descriptions.

Reviewer: Revert these changes to keep original. Just be sure to list all metrics in the metadata file.
```suggestion
See [metadata.csv][8] for a list of metrics provided by this check.
```

### Service Checks

See [service_checks.json][9] for a list of service checks provided by this integration.

> ### Events

Reviewer: Currently, this ### Events section only shows in the public docs page. I recommend moving this section under the ## Overview section in order to display both in-app and in the public docs.

> All Cloudsmith-related events collected appear within the Datadog Event Stream with the `source:cloudsmith` property. Events are collected every five minutes to reduce the number of requests sent to the Cloudsmith API.

Reviewer: ```suggestion
All Cloudsmith-related events collected appear within the Datadog Event Explorer with the `source:cloudsmith` tag. Events are collected every five minutes to reduce the number of requests sent to the Cloudsmith API.
```

There are several types of events available:

- Audit Logs event
- Security Scan event
- Vulnerability Policy Violation event
- License Policy Violation event
- Organization Members summary
- Quota (Raw Usage) Summary

They are accessible using the following aggregation keys:

> - `@aggregation_key:audit_log`
> - `@aggregation_key:vulnerabilities`
> - `@aggregation_key:vulnerability_policy_violation`
> - `@aggregation_key:license_policy_violation`
> - `@aggregation_key:org_members_summary`
> - `@aggregation_key:quota_summary`

Reviewer: Remove as this is no longer supported (https://docs.datadoghq.com/service_management/events/guides/migrating_to_new_events_features/#event-aggregations-are-no-longer-performed-or-displayed-in-the-ui)
```suggestion
```

> ## Troubleshooting

Reviewer: ```suggestion
## Support
```
